textile/features.textile

```diff
@@ -1305,6 +1307,23 @@ h4. Message
 ** @(TM2e)@ @encoding@ string
 ** @(TM2i)@ @extras@ JSON-encodable object, used to contain any arbitrary key value pairs which may also contain other primitive JSON types, JSON-encodable objects or JSON-encodable arrays. The @extras@ field is provided to contain message metadata and/or ancillary payloads in support of specific functionality, e.g. push. Each of these supported extensions is documented separately; for 1.1 the only supported extension is @push@, via the @extras.push@ member; 1.2 adds the @delta@ extension whose keys and values are described by the attributes of the type @DeltaExtras@, and the @headers@ extension, which contains arbitrary @string->string@ key-value pairs, settable at publish time, and @ref@ whose keys and values are described by the attributes of the type @ReferenceExtras@. Unless otherwise specified, the client library should not attempt to do any filtering or validation of the @extras@ field itself, but should treat it opaquely, encoding it and passing it to realtime unaltered.
 ** @(TM2f)@ @timestamp@ time in milliseconds since epoch. If a message received from Ably does not contain a @timestamp@, it should be set to the @timestamp@ of the encapsulating @ProtocolMessage@
+** @(TM2j)@ @action@ enum - the type of the message. Any message received from Ably must have an @action@ field set to one of the following values:
+*** @(TM2j1)@ 0: @MESSAGE_UNSET@
+*** @(TM2j2)@ 1: @MESSAGE_CREATE@
+*** @(TM2j3)@ 2: @MESSAGE_UPDATE@
+*** @(TM2j4)@ 3: @MESSAGE_DELETE@
+*** @(TM2j5)@ 4: @ANNOTATION_CREATE@
+*** @(TM2j6)@ 5: @ANNOTATION_DELETE@
+*** @(TM2j7)@ 6: @META_OCCUPANCY@
+** @(TM2K)@ @serial@ string - the origin timeserial of the message. This will be set, after publish, to the @channelSerial@ of the @ProtocolMessage@ that encapsulated the @Message@, with an additional offset that locates the @Message@ in the @ProtocolMessage@ messages array. If the message is received from Ably, this must be set.
+** @(TM2L)@ @refSerial@ string - if this message is an annotation that references another message, this must be set to the @serial@ of the referenced message. If the @refSerial@ is set, the @refType@ must also be set.
+** @(TM2M)@ @refType@ string - if the @refSerial@ is set, this must be set. If describes the type of the reference.
+** @(TM2N)@ @updatedAt@ time in milliseconds since epoch. If a message received from Ably has the @action@ field set to "MESSAGE_UPDATE" then the @updatedAt@ field must be set to the time the message was updated.
+** @(TM2O)@ @deletedAt@ time in milliseconds since epoch. If a message received from Ably has the @action@ field set to "MESSAGE_DELETE" then the @deleteAt@ field must be set to the time the message was deleted.
+** @(TM2P)@ @operation@ operation is an JSON-encodable object that contains the following `optional` fields;
```

Comment:
I don't see why we'd need to make this a JsonObject. It's a well-defined optional substruct of the Message with three known fields, why shouldn't the library  expose it accordingly? (In JS there might not be a difference (other than the typings), it could just be a plain object, but eg in java this would be much more usable as a MessageOperation class or whatever vs a JSONObject)